OpenAPI UI fails to load correctly

## Environment Details
* Helidon Version: 4.0.0
* Helidon MP
* JDK version: 21
* OS: Windows 11

----------

## Problem Description

**Expected behavior:** when the browser accesses http://localhost:8080/openapi/ui/index.html, the page has content. 
**Current behavior:** blank page.

There is no stack trace when using Firefox.
When using Chrome the logger writes:

> 2023.10.29 23:01:10 WARNING io.helidon.webserver.http.ErrorHandlers VirtualThread[#67,[0x05efff5a 0x1cecefd5] WebServer socket]/runnable@ForkJoinPool-1-worker-13: [0x05efff5a 0x1cecefd5] Request failed: HttpPrologue[protocol=HTTP, protocolVersion=1.1, method=GET, uriPath=/openapi/ui/swagger-ui-standalone-preset.js, query=, fragment=], cannot send error response, as response already sent
> io.helidon.http.RequestException: Cannot access static resource
>         at io.helidon.http@4.0.0/io.helidon.http.RequestException$Builder.build(RequestException.java:139)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.ErrorHandlers.unhandledError(ErrorHandlers.java:194)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:113)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.Filters$FilterChainImpl.proceed(Filters.java:121)
>         at io.helidon.webserver.observe.metrics@4.0.0/io.helidon.webserver.observe.metrics.MetricsFeature.lambda$configureVendorMetrics$1(MetricsFeature.java:90)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.Filters$FilterChainImpl.proceed(Filters.java:119)
>         at io.helidon.common.context@4.0.0/io.helidon.common.context.Contexts.runInContext(Contexts.java:117)
>         at io.helidon.webserver.context@4.0.0/io.helidon.webserver.context.ContextRoutingFeature.filter(ContextRoutingFeature.java:50)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.Filters$FilterChainImpl.proceed(Filters.java:119)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.Filters.executeFilters(Filters.java:87)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.Filters.lambda$filter$0(Filters.java:83)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:75)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.Filters.filter(Filters.java:83)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.HttpRouting.route(HttpRouting.java:109)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http1.Http1Connection.route(Http1Connection.java:332)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http1.Http1Connection.handle(Http1Connection.java:169)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.ConnectionHandler.run(ConnectionHandler.java:155)
>         at io.helidon.common.task@4.0.0/io.helidon.common.task.InterruptableTask.call(InterruptableTask.java:47)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.ThreadPerTaskExecutor$ThreadBoundFuture.run(ThreadPerTaskExecutor.java:239)
>         at java.base/java.lang.VirtualThread.run(VirtualThread.java:311)
> Caused by: io.helidon.http.InternalServerException: Cannot access static resource
>         at io.helidon.webserver.staticcontent@4.0.0/io.helidon.webserver.staticcontent.StaticContentHandler.handle(StaticContentHandler.java:232)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.HttpRouting$RoutingExecutor.doRoute(HttpRouting.java:668)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.HttpRouting$RoutingExecutor.call(HttpRouting.java:627)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.HttpRouting$RoutingExecutor.call(HttpRouting.java:605)
>         at io.helidon.webserver@4.0.0/io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:75)

I noticed that the files are downloaded by the browser, with the content inside, but they are all coming with the HTTP status `404`.
Thus it can be said it loads, but can't be said it does correctly.

## Steps to reproduce

1. Create a project using the Helidon MP Quick Start archetype.
2. Add the OpenAPI UI dependencies:
```xml
        <dependency>
            <groupId>io.helidon.integrations.openapi-ui</groupId>
            <artifactId>helidon-integrations-openapi-ui</artifactId>
            <scope>runtime</scope>
        </dependency>
        <dependency>
            <groupId>io.smallrye</groupId>
            <artifactId>smallrye-open-api-ui</artifactId>
            <version>3.7.0</version>
            <scope>runtime</scope>
        </dependency>
```

This issue is caused by the "Server Features" introduced in #7777. It effectively changed the HttpService registration order causing `OpenApiFeature` to be registered after `JaxRsService`.

`JaxRsService` catches to 404 errors from Jersey to continue the request chain, however the response status is still set to 404. The `StaticContentHandler` does not set the status code, and thus the response is sent with the proper payload but with a 404 status.

The index.html is served, but when loading the static resources the browser does not download the response payload which causes the blank page, see screenshot below.

![Screenshot 2023-11-21 at 1 55 06 PM](https://github.com/helidon-io/helidon/assets/3830708/248467c6-4f31-4313-a82f-72456ea6460b)

Also the non download of the response payload for the page resources causes broken pipe errors:
```
Caused by: java.io.UncheckedIOException: java.net.SocketException: Broken pipe
	at io.helidon.common.buffers.FixedBufferData.writeTo(FixedBufferData.java:74)
	at io.helidon.common.socket.PlainSocket.write(PlainSocket.java:136)
	at io.helidon.common.socket.SocketWriter.writeNow(SocketWriter.java:77)
	at io.helidon.common.socket.SocketWriterDirect.write(SocketWriterDirect.java:43)
	at io.helidon.webserver.http1.Http1ServerResponse$BlockingOutputStream.writeContent(Http1ServerResponse.java:680)
	at io.helidon.webserver.http1.Http1ServerResponse$BlockingOutputStream.write(Http1ServerResponse.java:567)
	at io.helidon.webserver.http1.Http1ServerResponse$BlockingOutputStream.write(Http1ServerResponse.java:438)
	at java.base/java.io.BufferedOutputStream.implWrite(BufferedOutputStream.java:217)
	at java.base/java.io.BufferedOutputStream.write(BufferedOutputStream.java:206)
	at java.base/java.io.InputStream.transferTo(InputStream.java:797)
	at java.base/sun.nio.ch.ChannelInputStream.transferTo(ChannelInputStream.java:268)
	at io.helidon.webserver.staticcontent.FileBasedContentHandler.send(FileBasedContentHandler.java:116)
	at io.helidon.webserver.staticcontent.CachedHandlerJar.handle(CachedHandlerJar.java:69)
	at io.helidon.webserver.staticcontent.ClassPathContentHandler.doHandle(ClassPathContentHandler.java:142)
	at io.helidon.webserver.staticcontent.StaticContentHandler.handle(StaticContentHandler.java:220)
	... 21 more
Caused by: java.net.SocketException: Broken pipe
	at java.base/sun.nio.ch.SocketDispatcher.write0(Native Method)
	at java.base/sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:62)
	at java.base/sun.nio.ch.NioSocketImpl.tryWrite(NioSocketImpl.java:394)
	at java.base/sun.nio.ch.NioSocketImpl.implWrite(NioSocketImpl.java:413)
	at java.base/sun.nio.ch.NioSocketImpl.write(NioSocketImpl.java:440)
	at java.base/sun.nio.ch.NioSocketImpl$2.write(NioSocketImpl.java:819)
	at java.base/java.net.Socket$SocketOutputStream.write(Socket.java:1195)
	at io.helidon.common.buffers.FixedBufferData.writeTo(FixedBufferData.java:71)
	... 35 more
```

When loading the page resources explicitly (E.g. with a separate tab) the browser actually downloads the response payload.

This change of behavior may impact any features or HttpService that doesn't set the status on the response.

At fist glance the fix seems to be to reset the status in JaxRsService when "nexting" the request:

```diff
diff --git a/microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java b/microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java
index 159bbf5b58..bdaeb5a413 100644
--- a/microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java
+++ b/microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java
@@ -256,6 +256,7 @@ class JaxRsService implements HttpService {
                 // reset may be impossible if data has already been written over the network
                 if (res instanceof RoutingResponse routing) {
                     if (routing.reset()) {
+                        res.status(Status.OK_200);
                         routing.next();
                     }
                 }
```

--- a/microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java
+++ b/microprofile/server/src/main/java/io/helidon/microprofile/server/JaxRsService.java
@@ -256,6 +256,7 @@ class JaxRsService implements HttpService {
                 // reset may be impossible if data has already been written over the network
                 if (res instanceof RoutingResponse routing) {
                     if (routing.reset()) {
+                        res.status(Status.OK_200);
                         routing.next();
                     }
                 }

